src/ucp/rndv/proto_rndv.c

```diff
@@ -668,8 +689,10 @@ void ucp_proto_rndv_receive_start(ucp_worker_h worker, ucp_request_t *recv_req,
     uint8_t sg_count;
     ucp_ep_h ep;
 
-    UCP_WORKER_GET_VALID_EP_BY_ID(&ep, worker, rts->sreq.ep_id, return,
-                                  "RTS on non-existing endpoint");
+    UCP_WORKER_GET_VALID_EP_BY_ID(&ep, worker, rts->sreq.ep_id, {
+        ucp_rndv_recv_req_complete(recv_req, UCS_ERR_CANCELED);
```

Comment:
IMO we should call request abort here for consistency
and better not reuse protov1 functions from protov2